private/buf/cmd/buf/buf.go

```diff
@@ -150,23 +148,36 @@ func NewRootCommand(name string) *appcmd.Command {
 				Use:   "mod",
 				Short: "Configure and update buf modules.",
 				SubCommands: []*appcmd.Command{
-					modinit.NewCommand("init", builder, "", false),
+					modinit.NewCommand("init", builder, modInitDeprecationMessage, true), // TODO(alex): Think about this one (see TODO below).
```

Comment:
`buf mod init` could be moved to `buf config init` and clearly state itself as "The behavior of this command is interactive, and is subject to change". That way, it can also support `buf.gen.yaml`, `buf.work.yaml`, or any combination of the three (or more later).